src/glfs-flock.c

```diff
@@ -0,0 +1,351 @@
+        int ret = -1;
+        int opt = 0;
+        int option_index = 0;
+        struct xlator_option *option;
+
+        optind = 0;
+        while (true) {
+                opt = getopt_long (argc, argv, "dehno:p:suv", long_options,
+                                   &option_index);
+
+                if (opt == -1) {
+                        break;
+                }
+
+                switch (opt) {
+                        case 'd':
+                                state->debug = true;
+                                break;
+                        case 'e':
+                                state->l_type = F_WRLCK;
+                                break;
+                        case 'n':
+                                state->block = false;
+                                break;
+                        case 'o':
+                                option = parse_xlator_option (optarg);
+                                if (option == NULL) {
+                                        error (0, errno, "%s", optarg);
+                                        goto err;
+                                }
+
+                                if (append_xlator_option (&state->xlator_options, option) == -1) {
+                                        error (0, errno, "append_xlator_option: %s", optarg);
+                                        goto err;
+                                }
+
+                                break;
+                        case 'p':
+                                port = strtoport (optarg);
+                                if (port == 0) {
+                                        goto out;
+                                }
+
+                                break;
+                        case 's':
+                                state->l_type = F_RDLCK;
+                                break;
+                        case 'u':
+                                state->l_type = F_UNLCK;
+                                break;
+                        case 'v':
+                                printf ("%s (%s) %s\n%s\n%s\n%s\n",
+                                        program_invocation_name,
+                                        PACKAGE_NAME,
+                                        PACKAGE_VERSION,
+                                        COPYRIGHT,
+                                        LICENSE,
+                                        AUTHORS);
+                                ret = -2;
+                                goto out;
+                        case 'h':
+                                usage ();
+                                ret = -2;
+                                goto out;
+                        default:
+                                goto err;
+                }
+        }
+
+        if (argc <= optind) {
+                usage ();
+                ret = -2;
+                goto out;
+        } else {
+                state->url = strdup (argv[argc - 1]);
```

Comment:
similarly, if we've already reached this point, we know that we have an active connection at this point, so most of this code can be removed.

from `do_flock`:

```c
if (ctx->fs) {
        ret = parse_options (argc, argv, true);
```